paddle/fluid/framework/parallel_executor.cc

```diff
@@ -63,7 +64,17 @@ ParallelExecutor::ParallelExecutor(
   member_->global_scope_ = scope;
   member_->use_cuda_ = exec_strategy.use_cuda_;
 
-  // Step 1. Bcast the params to devs.
+  // Step 1. check all memory usages of all places.
+  // This step will create BuddyAllocator for each place, which will
+  //    1. enforce that the place is avaliable and NOT used.
+  //    2. avoid ncclBcast hanging up for NOT enough memory to use.
+  for (size_t i = 0; i < member_->places_.size(); ++i) {
+    size_t usage = memory::memory_usage(member_->places_[i]);
+    VLOG(4) << "Memory usage of device: " << member_->places_[i] << " is "
+            << usage << " bytes";
+  }
```

Comment:
This check just should do one time. So maybe it is not appropriate for placing it here.

Reply:
Because BuddyAllocator's array was a static member in function in this [line](https://github.com/PaddlePaddle/Paddle/blob/develop/paddle/fluid/memory/malloc.cc#L62), so this check can do many times in the constructor in  ParallelExecutor